opennebula/resource_opennebula_virtual_network.go

```diff
@@ -1694,15 +1694,16 @@ func resourceOpennebulaVirtualNetworkDelete(ctx context.Context, d *schema.Resou
 	}
 	log.Printf("[INFO] Successfully released reservered IP addresses.")
 
-	err = vnc.Delete()
-	if err != nil {
-		diags = append(diags, diag.Diagnostic{
-			Severity: diag.Error,
-			Summary:  "Failed to delete",
-			Detail:   fmt.Sprintf("virtual network (ID: %s): %s", d.Id(), err),
-		})
-		return diags
-	}
+	err = resource.RetryContext(ctx, d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
```

Comment:
Did you ensured that prepending the Delete call with a `waitForVNetworkState` call wouldn't fix this ?
As a side note, you're using TimeoutCreate in the delete method here

Reply:
It does in the tests. This was explicitly a problem with floating IPs if I remember. It happens that ON takes a little more time to release floating IPs and it was creating test failures (see one of such tests here https://github.com/Thiryn/terraform-provider-opennebula/actions/runs/10548061643/job/29221515518#step:10:575).

In reality, this fix should also be applied to the AR resources, but this isn't the point of this PR. The only reason the fix is limited to the network, is because we use the now-deprecated feature of AR entries in the vnet resources in our VR tests https://github.com/OpenNebula/terraform-provider-opennebula/blob/master/opennebula/resource_opennebula_virtual_router_test.go#L285-L289

Reply:
For the timeout, I actually just copied it from somewhere else, I haven't looked at the deeper logic, looks like it is used in many other places that way https://github.com/search?q=repo%3AOpenNebula%2Fterraform-provider-opennebula%20d.Timeout(schema.TimeoutCreate)&type=code